pkg/controller/erspan.go

```diff
@@ -0,0 +1,386 @@
+			},
+			DeleteFunc: func(obj interface{}) {
+				cont.erspanPolicyDelete(obj)
+			},
+		},
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+	)
+	cont.log.Debug("Initializing Erspan Policy Informers")
+}
+
+func (cont *AciController) initPodIfInformerBase(listWatch *cache.ListWatch) {
+	cont.podIfIndexer, cont.podIfInformer = cache.NewIndexerInformer(
+		listWatch, &podIfpolicy.PodIF{}, 0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				cont.podIFAdded(obj)
+			},
+			UpdateFunc: func(oldobj interface{}, newobj interface{}) {
+				cont.podIFAdded(newobj)
+			},
+			DeleteFunc: func(obj interface{}) {
+				cont.podIFDeleted(obj)
+			},
+		},
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+	)
+	cont.log.Debug("Initializing podif Informers")
+}
+
+func (cont *AciController) getPodifKey(podif *podIfpolicy.PodIF) string {
+	return podif.Status.PodNS + "/" + podif.Status.PodName
+}
+
+func (cont *AciController) podIFAdded(obj interface{}) {
+	podif := obj.(*podIfpolicy.PodIF)
+	podifkey, err := cache.MetaNamespaceKeyFunc(podif)
+	if err != nil {
+		PodIfPolicyLogger(cont.log, podif).
+			Error("Could not create key:" + err.Error())
+		return
+	}
+	cont.log.Debug("podif key: ", podifkey)
+	cont.log.Infof("podif Added: %s", podif.ObjectMeta.Name)
+	cont.indexMutex.Lock()
+	deconcatenateEpg := strings.Split(podif.Status.EPG, "|")
+	deconcatenatedEpg := deconcatenateEpg[len(deconcatenateEpg)-1]
+	podifdata := &EndPointData{
+		MacAddr:   podif.Status.MacAddr,
+		EPG:       deconcatenatedEpg,
+		Namespace: podif.Status.PodNS,
+	}
+	podifKey := cont.getPodifKey(podif)
+	cont.podIftoEp[podifKey] = podifdata
+	cont.indexMutex.Unlock()
+	cont.queuePodIfUpdateByKey(podifkey)
+}
+
+func (cont *AciController) podIFDeleted(obj interface{}) {
+	podif := obj.(*podIfpolicy.PodIF)
+	podifkey, err := cache.MetaNamespaceKeyFunc(podif)
+	if err != nil {
+		PodIfPolicyLogger(cont.log, podif).
+			Error("Could not create key:" + err.Error())
+		return
+	}
+	cont.log.Infof("podif Deleted: %s", podif.ObjectMeta.Name)
+	cont.indexMutex.Lock()
+	delete(cont.podIftoEp, podif.ObjectMeta.Name)
+	cont.indexMutex.Unlock()
+	cont.queuePodIfUpdateByKey(podifkey)
+}
+
+func (cont *AciController) erspanPolicyUpdated(obj interface{}) {
+	erspanPolicy := obj.(*erspanpolicy.ErspanPolicy)
+	spankey, err := cache.MetaNamespaceKeyFunc(erspanPolicy)
+	if err != nil {
+		ErspanPolicyLogger(cont.log, erspanPolicy).
+			Error("Could not create key:" + err.Error())
+		return
+	}
+	cont.erspanPolPods.UpdateSelectorObj(obj)
+	cont.queueErspanUpdateByKey(spankey)
+
+}
+
+func (cont *AciController) queueErspanUpdateByKey(key string) {
+	cont.log.Debug("erspan key queued: ", key)
+	cont.erspanQueue.Add(key)
+}
+
+func (cont *AciController) queuePodIfUpdateByKey(key string) {
+	cont.log.Debug("Podif key queued: ", key)
+	cont.podIfQueue.Add(key)
+}
+
+func (cont *AciController) handlePodifInfo(podif *podIfpolicy.PodIF) bool {
```

Comment:
Do we really need this? It looks like this does a lookup and logs differently based on whether podif is already in the map or not...that doesn't seem worth the handling. Can we remove this or am I missing something?

Reply:
We need to add podif obj to the queue in the environment.go and process it. Otherwise the podif crd wont be processed in the controller. We can remove the remove the if else statements and have an empty func.

Reply:
Are you sure about that? I don't see why you need the queue at all. Your callback isn't doing anything and I don't see any side effects in the processQueue handling that would be needed. The calls to queuePodIfUpdateByKey also don't seem to be needed

Reply:
we don't require a queue here this can be removed. you can handle it directly Podif update function